### PR TITLE
Replace CircleCI and Travis tests with Github actions

### LINKED ISSUE #1615: Since we are out of travis builds, we should probably move to GitHub actions
Currently, travis CI does not run.

### PR DESCRIPTION
**Pull request by jasongrout**
This moves us over to Github Actions for the CI, as we've been doing with many Jupyter projects.

Fixes #1615

### CODE REVIEW

**Review by MSeal — APPROVED** [review]
Thanks, that was a task on the back burner for this repo. Glad to see it getting updated

**Comment by SylvainCorlay** [comment]
No no, travisci.

We run the tests on travis, but we are out of builds. Circle only builds the docs.

**Comment by jasongrout** [comment]
Oh, hey, there *is* a .travis yml file :) https://github.com/jupyter/nbconvert/blob/main/.travis.yml

**Comment by jasongrout** [comment]
Tests are passing now without any skipped tests (needed to install inkscape and xelatex).

File: docs/source/nbconvert_library.ipynb
Comment (jasongrout) [inline docs/source/nbconvert_library.ipynb:50]: This change shouldn't have to be made. It's only here because the link check things that it needs to check this code, and gets the link wrong.

File: docs/source/nbconvert_library.ipynb
Comment (jasongrout) [inline docs/source/nbconvert_library.ipynb:82]: This change shouldn't have to be made. It's only here because the link check gets the link wrong.

Comment (jasongrout) [reply docs/source/nbconvert_library.ipynb]: This is now fixed by using nbsphinx properly in 3832b3c

File: docs/source/nbconvert_library.ipynb
Comment (jasongrout) [reply docs/source/nbconvert_library.ipynb]: This is now fixed by using nbsphinx properly in 3832b3c

**Comment by jasongrout** [comment]
Presumably the CircleCI tests are failing because I deleted those files. After this is merged, I'll delete the integration.

**Comment by jasongrout** [comment]
I think this is ready for review. @SylvainCorlay @MSeal

**Comment by jasongrout** [comment]
I just noticed that this repo has merge commits disabled on merging PRs (so only squash and merge or rebase and merge). I think this is the first time I've seen merge commits disabled on a Jupyter repo. I'm curious: when was the repo setting changed to disable merge commits?

**Review by MSeal — APPROVED** [review]
Thanks!

**Comment by MSeal** [comment]
@jasongrout I can't find the issue were it was discussed. But someone wanted it set that way during 6.0 development because some PRs with a lot of patch commits would have dominated the history. I use Squash and Merge in my work repos and didn't see an issue changing it to that pattern. Saves asking people to cleanup commit histories, but I'd good with going with whatever the larger team wants to use if there's consensus or strong opinions.

**Comment by jasongrout** [comment]
> I use Squash and Merge in my work repos and didn't see an issue changing it to that pattern. Saves asking people to cleanup commit histories, but I'd good with going with whatever the larger team wants to use if there's consensus or strong opinions.

Since you're the principal maintainer here, I'm happy to defer to your judgement (I don't think we have a project-wide policy). I personally rarely use squash and commit because I like having the extra context in history (i.e., when commits are squashed, the commit messages are appended to each other, so you lose some context about how specific changes to specific lines relate to specific comments that go with that commit).  But really, whatever you want here. I realize it's personal preference either way and there are strong arguments for either approach.

**Comment by jasongrout** [comment]
Just to be clear on process: are you planning on merging this @MSeal, or are you waiting for another approval, or am I expected to merge it now that you approved it?

**Comment by MSeal** [comment]
Ahh for other maintainers of Jupyter, once approved use your own discretion about merging and format of merging. I can merge now as I normally would with a PR review.

**Comment by jasongrout** [comment]
I've deactivated the travis and circleci webhooks (as a test to make sure nothing goes wrong before deleting the webhooks).

**Comment by jasongrout** [comment]
> I've deactivated the travis and circleci webhooks (as a test to make sure nothing goes wrong before deleting the webhooks).

I just deleted the travis and circleci webhooks permanently.

**Pull request merged**
